### PR TITLE
add parameters for postprocessing with tailor

### LINKED ISSUE #357: new parameter functions for postprocessing with tailor
Opening in favor of https://github.com/tidymodels/tailor/issues/7. A few tailor adjustments register call info for dials functions that don't yet exist:

* `adjust_equivocal_zone()`: `buffer()`
* `adjust_numeric_range()`: `lower_limit()`, `upper_limit()`

`adjust_probability_threshold()` references `threshold()`, which does exist already.

### PR DESCRIPTION
**Pull request by simonpcouch**
Closes #357!

### CODE REVIEW

File: R/param_range_limits.R [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#' @name range_limits
+#' @export
+
+#' @rdname range_limits
+#' @export
+lower_limit <- function(range = c(-Inf, Inf), trans = NULL) {
+  new_quant_param(
+    type = "double",
+    range = range,
+    inclusive = c(TRUE, FALSE),
```
Comment (simonpcouch) [inline R/param_range_limits.R:19]: A second set of eyes on this for Wonk Prevention would be appreciated. We allow lower limits from [-Inf, Inf)—i.e. one can't set a lower limit of infinity.

File: tests/testthat/test-params.R [hunk 2]
```diff
@@ -4,6 +4,7 @@ test_that("param ranges", {
   expect_equal(sample_size(1:2)$range, list(lower = 1L, upper = 2L))
   expect_equal(learn_rate(c(.1, .9))$range, list(lower = 0.1, upper = 0.9))
   expect_equal(loss_reduction(c(.1, .9))$range, list(lower = 0.1, upper = 0.9))
+  expect_equal(buffer(c(0, .25))$range, list(lower = 0, upper = .25))
```
Comment (simonpcouch) [inline tests/testthat/test-params.R:7]: Tried to plop these in reasonable places but wasn't clear to me whether I missed some ordering logic.

**Review by topepo — APPROVED** [review]

File: R/param_range_limits.R (hunk 1 above)
Comment (hfrick) [reply R/param_range_limits.R:19]: That sounds resonable to me

File: tests/testthat/test-params.R (hunk 2 above)
Comment (hfrick) [reply tests/testthat/test-params.R:7]: I think the order is largly but maybe not exclusively "historically grown"

**Review by hfrick — APPROVED** [review]

**Review by hfrick — APPROVED** [review]

**Pull request merged**